sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/util/CharVarcharCodegenUtils.java

```diff
@@ -22,55 +22,34 @@
 public class CharVarcharCodegenUtils {
   private static final UTF8String SPACE = UTF8String.fromString(" ");
 
-  /**
-   *  Trailing spaces do not count in the length check. We don't need to retain the trailing
-   *  spaces, as we will pad char type columns/fields at read time.
-   */
-  public static UTF8String charTypeWriteSideCheck(UTF8String inputStr, int limit) {
-    if (inputStr == null) {
-      return null;
+  private static UTF8String trimTrailingSpaces(
+      UTF8String inputStr, int numChars, int limit) {
+    int numTailSpacesToTrim = numChars - limit;
```

Comment:
nit: to be safe we should add `assert(numChars > limit)`

Reply:
to be safer we can add this assert to  UTF8String

Reply:
yea we can add it in `trimTrailingSpaces`